tests/test_device.py

```diff
@@ -43,3 +44,119 @@ def test_binary_division():
     assert dev_color.hex_color == 'f1e0b5'
     assert dev_color.xy_color == (30015, 26870)
+
+
+def test_set_hsb():
+    dev = Device(LIGHT_WS)
+
+    command = dev.light_control.set_hsb(300, 200)
+    data = command.data[ATTR_LIGHT_CONTROL][0]
+    assert len(data) is 2
+
+    command = dev.light_control.set_hsb(300, 200, None)
+    data = command.data[ATTR_LIGHT_CONTROL][0]
+    assert len(data) is 2
+
+    command = dev.light_control.set_hsb(300, 200, 100)
+    data = command.data[ATTR_LIGHT_CONTROL][0]
+    assert len(data) is 3
+
+    command = dev.light_control.set_hsb(300, 200, 100, transition_time=1)
+    data = command.data[ATTR_LIGHT_CONTROL][0]
+    assert len(data) is 4
+
+
+def test_set_xy_color():
+    dev = Device(LIGHT_WS)
+
+    command = dev.light_control.set_xy_color(300, 200)
+    data = command.data[ATTR_LIGHT_CONTROL][0]
+    assert len(data) is 2
+
+    command = dev.light_control.set_xy_color(300, None)
+    data = command.data[ATTR_LIGHT_CONTROL][0]
+    assert len(data) is 2
+
+    command = dev.light_control.set_xy_color(300, 200, transition_time=1)
+    data = command.data[ATTR_LIGHT_CONTROL][0]
+    assert len(data) is 3
+
+
+def test_set_color_temp():
+    dev = Device(LIGHT_WS)
+
+    command = dev.light_control.set_color_temp(300)
+    data = command.data[ATTR_LIGHT_CONTROL][0]
+    assert len(data) is 1
+
+    command = dev.light_control.set_color_temp(None)
+    data = command.data[ATTR_LIGHT_CONTROL][0]
+    assert len(data) is 1
+
+    command = dev.light_control.set_color_temp(300, transition_time=1)
+    data = command.data[ATTR_LIGHT_CONTROL][0]
+    assert len(data) is 2
+
+
+def test_set_dimmer():
+    dev = Device(LIGHT_WS)
+
+    command = dev.light_control.set_dimmer(200)
+    data = command.data[ATTR_LIGHT_CONTROL][0]
+    assert len(data) is 1
+
+    command = dev.light_control.set_dimmer(None)
+    data = command.data[ATTR_LIGHT_CONTROL][0]
+    assert len(data) is 1
+
+    command = dev.light_control.set_dimmer(200, transition_time=1)
+    data = command.data[ATTR_LIGHT_CONTROL][0]
+    assert len(data) is 2
+
+
+def test_set_state():
+    dev = Device(LIGHT_WS)
+
+    with pytest.raises(TypeError):
+        dev.light_control.set_state(None)
+
+    command = dev.light_control.set_state(True)
+    data = command.data[ATTR_LIGHT_CONTROL][0]
+    assert len(data) is 1
+
+    command = dev.light_control.set_state(False)
+    data = command.data[ATTR_LIGHT_CONTROL][0]
+    assert len(data) is 1
+
+
+def test_set_hex_color():
+    dev = Device(LIGHT_WS)
+
+    command = dev.light_control.set_hex_color("4a418a")
+    data = command.data[ATTR_LIGHT_CONTROL][0]
+    assert len(data) is 1
+
+    command = dev.light_control.set_hex_color("4a418a", transition_time=1)
+    data = command.data[ATTR_LIGHT_CONTROL][0]
+    assert len(data) is 2
+
+    command = dev.light_control.set_hex_color("RandomString")
+    data = command.data[ATTR_LIGHT_CONTROL][0]
+    assert len(data) is 1
```

Comment:
Is it a problem that every string is accepted here? Perhaps we should limit the string length to 6, that could prevent accidental network overload.

Reply:
Not needed now, let's do it in another PR late ron.